### ISSUE TITLE
Minimap map not visible

### ISSUE THREAD

**Issue by jukos** [andreaordonselli/qgis2opengis#4]
Hello,
I have replaced the minimap.png with my own in the resources folder. The png was created with QGIS: Export map as image. 
What else do I have to do so that I can see the map? I also have a PGW file. Where can I set the coordinates for the image?
The navigation within the minimap works.

QGIS 3.34.6
qgis2opengis 2.20
Test on server; change in export folder.

![minimap](https://github.com/andreaordonselli/qgis2opengis/assets/31866672/6c66f4f4-b3ce-47ce-a01b-6fb519baf24e)

**Comment by andreaordonselli** [comment]
It's a bug that I will fix, for now edit the index.html file, scroll to the bottom and edit this part of code.
Acquiring the image from qgis is the best practice.
Let me know if you solve it.
Greetings.


```
var MaxMapExtent = map.getView().getProperties().extent;
			var minZoom = map.getView().getMinZoom();
			var maxZoom = map.getView().getMaxZoom();
			var ov = new ol.control.Overview({
			  layers: [ 
				new ol.layer.Image({
					source: new ol.source.ImageStatic({
						url: './resources/minimap.png',
						imageExtent: MaxMapExtent,
					})
				})
			  ],
			  minZoom: minZoom,
			  maxZoom: maxZoom,
			  rotation: true,
			  align: 'bottom-left',
			  //panAnimation: "elastic" 
			});
			map.addControl(ov);	
			// Select button
			var overviewbuttontitle = $('div.ol-overview button');
			// Set button title
			overviewbuttontitle.attr('title', 'Minimap');
```

**Comment by jukos** [comment]
Hello,
yes, it's better with the new code.
However, the display only works with the export setting “Restrict to extent”. The problem is that at minimum zoom the png map looks so small within the minimap box. How could I change this?
Greetings

![minimap2](https://github.com/andreaordonselli/qgis2opengis/assets/31866672/441e94d0-c9d6-4b82-9312-428f57b35ecf)

**Comment by andreaordonselli** [comment]
Yes, I confirm that if "restrict to extent" is not defined it is not possible to use the minimap with static image.
This is because the map itself does not have a maximum extent, so we cannot apply it to the minimap.

For the issue of the small image, manually increase the value of minZoom.
Write minZoom: 1, then minZoom: 2 until you find your right value.
Once you have found the correct value you can set "Min Zoom level" during export
![image](https://github.com/andreaordonselli/qgis2opengis/assets/89784373/631e49f8-d16c-408f-a8ec-1e78f8476bab)

An alternative to all this, which always works, both with "restrict to extenet" and without, and it is not necessary to set minZoom or maxZoom, would be not to use a fixed image but an external layer such as OSM. Would it suit you?

Be careful though: when using a third-party layer you must follow the manager's guidelines. Some want to be mentioned, others don't want their basemaps to be overlaid with other data, etc.

For examplem here the code for OpenStreetMap:
```javascript
var ov = new ol.control.Overview({
  layers: [ 
    new ol.layer.Tile({
      source: new ol.source.OSM()
    }),
  ],
  rotation: true,
  align: 'bottom-left',
  //panAnimation: "elastic"
});
map.addControl(ov);
// Select button
var overviewbuttontitle = $('div.ol-overview button');
// Set button title
overviewbuttontitle.attr('title', 'Minimap');
```
or OpenTopoMap:
```javascript
var ov = new ol.control.Overview({
  layers: [ 
    new ol.layer.Tile({
      source: new ol.source.XYZ({
        url: 'https://tile.opentopomap.org/{z}/{x}/{y}.png'
      })
    }),
  ],
  rotation: true,
  align: 'bottom-left',
  //panAnimation: "elastic"
});
map.addControl(ov);
// Select button
var overviewbuttontitle = $('div.ol-overview button');
// Set button title
overviewbuttontitle.attr('title', 'Minimap');
```
or Google Satellite

```javascript
var ov = new ol.control.Overview({
  layers: [ 
    new ol.layer.Tile({
      source: new ol.source.XYZ({
        url: 'https://mt1.google.com/vt/lyrs=s&x={x}&y={y}&z={z}'
      })
    }),
  ],
  rotation: true,
  align: 'bottom-left',
  //panAnimation: "elastic"
});
map.addControl(ov);
// Select button
var overviewbuttontitle = $('div.ol-overview button');
// Set button title
overviewbuttontitle.attr('title', 'Minimap');
```

**Comment by jukos** [comment]
Hello,
thank you for your reply. Both solutions work.
Could I also add an own xyz tile layer? Could I then insert the URL under "source"? 
If so, could you please also write a sample code for this as well?

Thank you very much and greetings

**Comment by andreaordonselli** [comment]
You can enter any XYZ level, just enter it in the `url` of new `ol.source.XYZ({`

If you have layers already defined and exported in your project, you could also take the name of this layer by finding it in the `layers.js` file and inserting it into the minimap code. You notice them because they all start with `lyr_ecc...`
For example

```javascript
var ov = new ol.control.Overview({
  layers: [lyr_layername1,lyr_layername2],
  rotation: true,
  align: 'bottom-left',
  //panAnimation: "elastic"
});
map.addControl(ov);
// Select button
var overviewbuttontitle = $('div.ol-overview button');
// Set button title
overviewbuttontitle.attr('title', 'Minimap');
```

**Comment by andreaordonselli** [comment]
From your screen I noticed that you inserted svg icons. Great!
In the next versions the color and border of the SVGs will be supported (obviously only for hybrid svg which can be set in qgis).

**Comment by andreaordonselli** [comment]
Thanks for the tip.  I will update the documentation with the various methods that can be used.

**Comment by andreaordonselli** [comment]
This issue has been resolved in the v2.22.0 release
Thanks for your report
